lib/govuk_design_system_formbuilder/elements/date.rb

```diff
@@ -105,7 +122,8 @@ def input(segment, link_errors, width, value)
           pattern: pattern(segment),
           inputmode: 'numeric',
           value: value,
-          autocomplete: date_of_birth_autocomplete_value(segment)
+          autocomplete: date_of_birth_autocomplete_value(segment),
+          maxlength: maxlength,
```

Comment:
I think the `if`/`else` statements could be avoided by adding a `MAX_LENGTHS` constant, along the lines of

```
MAX_LENGTHS = { day: 2, month: 2, year: 4 }.freeze
```

then here if it falls back to `nil` the attribute will automatically be omitted by Rails:

```suggestion
          maxlength: max_length(segment),
        end
        
        def max_length(segment)
          return unless maxlength_enabled?
          
          MAX_LENGTHS.fetch(segment)
        end
```